configuration-baselines/.htaccess-joomla

```diff
@@ -0,0 +1,192 @@
+	ExpiresByType x-world/x-vrml "now plus 1 month"
+	ExpiresByType video/x-msvideo "now plus 1 month"
+	ExpiresByType video/mpeg "now plus 1 month"
+	ExpiresByType video/mp4 "now plus 1 month"
+	ExpiresByType video/quicktime "now plus 1 month"
+	ExpiresByType video/x-la-asf "now plus 1 month"
+	ExpiresByType video/x-ms-asf "now plus 1 month"
+</IfModule>
+
+# Disable caching of administrator/index.php
+<Files "administrator/index.php">
+	<IfModule mod_expires.c>
+		ExpiresActive Off
+	</IfModule>
+	<IfModule mod_headers.c>
+		Header unset ETag
+		Header set Cache-Control "max-age=0, no-cache, no-store, must-revalidate"
+		Header set Pragma "no-cache"
+		Header set Expires "Wed, 17 Aug 2005 00:00:00 GMT"
+	</IfModule>
+</Files>
+###########################################
+
+###########################################
+# == Automatic compression of resources ==
+<IfModule mod_deflate.c>
+	AddOutputFilterByType DEFLATE text/html text/plain text/xml text/css application/xml application/xhtml+xml application/rss+xml application/javascript application/x-javascript
+</IfModule>
+<IfModule mod_gzip.c>
+	mod_gzip_on Yes
+	mod_gzip_dechunk Yes
+	mod_gzip_keep_workfiles No
+	mod_gzip_can_negotiate Yes
+	mod_gzip_add_header_count Yes
+	mod_gzip_send_vary Yes
+	mod_gzip_min_http 1000
+	mod_gzip_minimum_file_size 300
+	mod_gzip_maximum_file_size 512000
+	mod_gzip_maximum_inmem_size 60000
+	mod_gzip_handle_methods GET
+	mod_gzip_item_include file \.(html?|txt|css|js|php|pl|xml|rb|py)$
+	mod_gzip_item_include mime ^text/plain$
+	mod_gzip_item_include mime ^text/html$
+	mod_gzip_item_include mime ^text/xml$
+	mod_gzip_item_include mime ^text/css$
+	mod_gzip_item_include mime ^application/xml$
+	mod_gzip_item_include mime ^application/xhtml+xml$
+	mod_gzip_item_include mime ^application/rss+xml$
+	mod_gzip_item_include mime ^application/javascript$
+	mod_gzip_item_include mime ^application/x-javascript$
+	mod_gzip_item_exclude rspheader ^Content-Encoding:.*gzip.*
+	mod_gzip_item_include handler ^cgi-script$
+	mod_gzip_item_include handler ^server-status$
+	mod_gzip_item_include handler ^server-info$
+	mod_gzip_item_include handler ^application/x-httpd-php
+	mod_gzip_item_exclude mime ^image/.*
+</IfModule>
+###########################################
+
+###########################################
+# ===== Advanced server protection  =======
+## Disable PHP Easter Eggs
+RewriteCond %{QUERY_STRING} \=PHP[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12} [NC]
+RewriteRule .* - [F]
+
+#### Front-end protection
+## Allow limited access for certain directories with client-accessible content
+RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts|cache/com_comment/init)/.*\.(jpe|jpg|jpeg|jp2|jpe2|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot|JPG|JPEG|PNG|GIF|CSS|JS|TTF|WOFF|EOT)$ - [L]
```

Comment:
1. What extensions are creating publicly accessible resources in the `cache` directory?  AFAIK there is no "com_comment" on any .org site, and whitelisting paths in the `cache` directory raises a red flag about extension code.

2. SVG should be allowed here.

3. You have some file extensions duplicated as all caps and lowercased.  It'd probably be good if the extension list were case insensitive so you're not duplicating the list.

Reply:
I don't know Regex the rest is done.
I found the following on Google
<img width="1435" alt="image" src="https://user-images.githubusercontent.com/11913918/83875864-293b9800-a738-11ea-9be8-b996d3f5a94f.png">

Reply:
As I understand we then should change:
`RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts)/.*\.(jpe|jpg|jpeg|jp2|jpe2|svg|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot|JPG|JPEG|PNG|GIF|CSS|JS|TTF|WOFF|EOT)$ - [L]`

Too
`RewriteRule ^(components|modules|templates|images|plugins|media|libraries|media/jui/fonts)/.*\.(jpe|jpg|jpeg|jp2|jpe2|svg|png|gif|bmp|css|js|swf|html|mpg|mp3|mpeg|mp4|avi|wav|ogg|ogv|xls|xlsx|doc|docx|ppt|pptx|zip|rar|pdf|xps|txt|7z|svg|odt|ods|odp|flv|mov|ico|htm|ttf|woff|eot)$ - [L,NC]`

Reply:
@mbabker do you have any opinion about my last message?